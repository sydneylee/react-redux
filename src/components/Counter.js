import React, {Component} from 'react';
import PropTypes from 'prop-types';

class Counter extends Component {
    static defaultProps = {
        number : -1
    };

    static propTypes = {};

    state = {
        number:0
    };

    handleOnIncrement= ()=>{
        this.setState({
            number: this.state.number +1,
        });

    };
    handleOnDecrement= ()=>{
        this.setState({
            number: this.state.number -1,
        });
    };
    render() {
        //const {number, onIncrement, onDecrement} = this.props;
        // return (
        //     <div>
        //         <h1>{number}</h1>
        //         <button onClick={onIncrement}>증가 (+)</button>
        //         <button onClick={onDecrement}>감소 (-)</button>
        //     </div>
        // );


        const {number} = this.state;

        return (
            <div>
                <h1>{number}</h1>
                <button onClick={this.handleOnIncrement}>증가 (+)</button>
                <button onClick={this.handleOnDecrement}>감소 (-)</button>
            </div>
        );
    }
}

export default Counter;
