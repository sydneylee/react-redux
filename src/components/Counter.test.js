// // 1) JEST default tesing - component rendering
// import React from 'react';
// import reactDOM from 'react-dom';
// import Counter from './Counter';
//
//
// it('render successfully', ()=>{
//    const div = document.createElement('div');
//     reactDOM.render(<Counter />, div);
//     reactDOM.unmountComponentAtNode(div);
// });


// // 2) using react-test-renderer :
// // 2-1) renderer.create(<Counter />):
// // 2-2) toMatchSnapshot();
// import React from 'react';
// //import reactDOM from 'react-dom';
// import renderer from 'react-test-renderer';
// import Counter from './Counter';
//
// describe('Counter', ()=>{
//     let component = null;
//     it('render successfully', ()=>{
//         // const div = document.createElement('div');
//         // reactDOM.render(<Counter />, div);
//         component = renderer.create(<Counter />);
//         //reactDOM.unmountComponentAtNode(div);
//     });
//
//     it('matches a rendered snapshot', ()=>{
//         const tree = component.toJSON();
//         expect(tree).toMatchSnapshot();
//     });
//
//     it('increase correctly', ()=>{
//         component.getInstance().handleOnIncrement();
//         expect(component.getInstance().state.number).toBe(1);
//         const tree = component.toJSON();
//         expect(tree).toMatchSnapshot();
//     });
//
//     it('decrease correctly', ()=>{
//         component.getInstance().handleOnDecrement();
//         expect(component.getInstance().state.number).toBe(0);
//         const tree = component.toJSON();
//         expect(tree).toMatchSnapshot();
//     });
//
// });

// 3) enzyme - using shallow
import React from 'react';
import {shallow} from 'enzyme';
import Counter from './Counter';

describe('Counter', ()=>{

    let component = null;
    it('render successfully with enzyme shallow method', ()=>{
        component = shallow(<Counter />);
    });

    it('matches a snapshot with shallow component', ()=>{
       expect(component).toMatchSnapshot();
    });

    describe('find button from rendered component', ()=>{
        it('found button', ()=>{
            expect(component.find('button').exists()).toBe(true);
        });

        it('simulate onClick for triggering increment', ()=>{
           const mockedEvent ={
               target:{
                   value : null
               }
           };
           //TODO:sj - Method “simulate” is only meant to be run on a single node. 2 found instead. => use id for selector

            component.find('button#incBtn').simulate('click', mockedEvent);

            //component.find('button').simulate('click', mockEvent);
            // console.log(component.state().number);
            expect(component.state().number).toBe(1);
            component.find('button#incBtn').simulate('click', mockedEvent);
            expect(component.state().number).toBe(2);
        });

        it('simulate onClick for triggering decrement', ()=>{
            const mockedEvent ={
                target:{
                    value : null
                }
            };
            component.find('button#decBtn').simulate('click', mockedEvent);
            expect(component.state().number).toBe(1);
            component.find('button#decBtn').simulate('click', mockedEvent);
            expect(component.state().number).toBe(0);
        });

    });

});

