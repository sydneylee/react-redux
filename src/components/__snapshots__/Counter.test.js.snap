// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter matches a snapshot with shallow component 1`] = `
<div>
  <h1>
    0
  </h1>
  <button
    onClick={[Function]}
  >
    증가 (+)
  </button>
  <button
    onClick={[Function]}
  >
    감소 (-)
  </button>
</div>
`;
